tests/unit/test_cursorManager.py

```diff
@@ -66,6 +66,81 @@ def test_selectForwardThenSelBackward(self):
 		cm.script_selectWord_back(None) # "b" unselected, "a" selected
 		self.assertEqual(cm.selectionOffsets, (0, 1)) # "a" selected
 
+	def test_selectBackwardThenSelForward(self):
+		"""Test selecting backward, then selecting forward without unselecting.
+		Same as test_selectForwardThenSelBackward, but with reversed directions.
+		"""
+		cm = CursorManager(text="abc", selection=(2, 2)) # Caret at "c"
+		cm.script_selectCharacter_back(None) # "b" selected
+		cm.script_selectWord_forward(None) # "b" unselected, "c" selected
+		self.assertEqual(cm.selectionOffsets, (2, 3)) # "c" selected
+
+	def test_selectForwardThenSelBackwardThenUnsel(self):
+		"""Test selecting forward, then selecting backward without unselecting, then unselecting forward.
+		Depends on behavior tested by test_selectForwardThenSelBackward.
+		"""
+		cm = CursorManager(text="abc", selection=(1, 1)) # Caret at "b"
+		cm.script_selectCharacter_forward(None) # "b" selected
+		cm.script_selectWord_back(None) # "b" unselected, "a" selected
+		cm.script_selectCharacter_forward(None) # "a" unselected
+		self.assertEqual(cm.selectionOffsets, (1, 1)) # Caret at "b", no selection
+
+	def test_selectBackwardThenSelForwardThenUnsel(self):
+		"""Test selecting backward, then selecting forward without unselecting, then unselecting backward.
+		Same as test_selectForwardThenSelBackwardThenUnsel, but with reversed directions.
+		Depends on behavior tested by test_selectBackwardThenSelForward.
+		"""
+		cm = CursorManager(text="abc", selection=(2, 2)) # Caret at "c"
+		cm.script_selectCharacter_back(None) # "b" selected
+		cm.script_selectWord_forward(None) # "b" unselected, "c" selected
+		cm.script_selectCharacter_back(None) # "c" unselected
+		self.assertEqual(cm.selectionOffsets, (2, 2)) # Caret at "c", no selection
+

```

Comment:
There is no coverage for `selectBackwardThenUnselThenSelForward` but there is for `selectForwardThenUnselThenSelBackward`? 

Can these tests be broken down? Is there state that can only be achieved by using the script methods? For instance, instead of testing as in `test_selectBackwardThenSelForwardThenUnsel`, instead have three tests (using a naming scheme of given_do_expect:

### test_caretAtC_SelectCharBack_bSelected
```
cm = CursorManager(text="abc", selection=(2, 2)) # Caret at "c"
cm.script_selectCharacter_back(None) # "b" selected
self.assertEqual(cm.selectionOffsets, (1, 2)) # 'b' selected
```
### test_bSelected_selectWordForward_cSelected
```
cm = CursorManager(text="abc", selection=(1, 2)) #'b' selected
cm.script_selectWord_forward(None) # "b" unselected, "c" selected
self.assertEqual(cm.selectionOffsets, (2, 3)) # 'c' selected
```
### test_cSelected_selectCharacterBack_NoSelection
```
cm = CursorManager(text="abc", selection=(2, 3)) #'c' selected
cm.script_selectCharacter_back(None) # "c" unselected
self.assertEqual(cm.selectionOffsets, (2, 2)) # Caret at "c", no selection
```

Perhaps there is a reason we cant do this, like some state on CursorManager that we can not set through the constructor. But if we can I think this makes it easier to verify good test coverage.